analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -114,18 +114,15 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
 
         private IEnumerable<ExplodedNode> ProcessOperation(ExplodedNode node)
         {
-            var context = new SymbolicContext(symbolicValueCounter, node.Operation, node.State);
-            context = InvokeChecks(context, x => x.PreProcess);
-            if (context != null)
+            foreach (var preProcessed in InvokeChecks(new(symbolicValueCounter, node.Operation, node.State), x => x.PreProcess))
             {
-                context = EnsureContext(context, ProcessOperation(context));
-                context = InvokeChecks(context, x => x.PostProcess);
-                if (context != null)
+                var processed = EnsureContext(preProcessed, ProcessOperation(preProcessed));
+                foreach (var postProcessed in InvokeChecks(processed, x => x.PostProcess))
```

Comment:
Can be simplified with` .Select(x => x.State)`

Reply:
I intentionally closed that issue on SC, because it would make the whole method hard to follow due to the `preProcessed`, `processed`, `postProcessed` logical sequence. It would be confusing to change the type for the last one.